### PR TITLE
Add method to select Ensemble time series with a particular version

### PR DESCRIPTION
**Pull request by danhamill**
- Added VersionIdentifier class
- Changed database.sql to include version in the unique index 
- Added unit test and database resource
Addresses #172

### CODE REVIEW

File: FIRO_TSEnsembles/src/main/resources/database.sql
Comment (ktarbet) [inline FIRO_TSEnsembles/src/main/resources/database.sql:1]: I like that you are adding version to this index to prevent inconsistent states.

Instead of updating the 'base' database.sql, Please try creating an resources/update_20210922_20230718.sql with the index update.  The script probably needs to drop the index (if it exists) then add back with your improvement.  This allows existing databases, and newly created databases,  to be upgraded to a consistent version.

There is also a versions.txt file that needs the new entry, and a line such as:

```sql
UPDATE version set version = '20230718';
```

File: FIRO_TSEnsembles/src/main/java/hec/SqliteDatabase.java [hunk 1]
```diff
@@ -328,6 +341,33 @@ private EnsembleTimeSeries readEnsembleTimeSeriesFromDB(RecordIdentifier timeser
         }
         return rval;
     }
+    private EnsembleTimeSeries readEnsembleTimeSeriesFromDB(VersionIdentifier versionID, String sql) {
```
Comment (ktarbet) [inline FIRO_TSEnsembles/src/main/java/hec/SqliteDatabase.java:344]: Please consider ways to reduce the duplicate code in the two private readEnsembleTimeSeriesFromDB(..) methods.   For example the code the loops through the ResultSet could be extracted to a shared method.  Another approach could be to pass the PreparedStatement to a shared readEnsembleTimeSeriesFromDB(preparedStatement) that handles both cases for  RecordIdentifier and VersionIdentifier.

**Review by ktarbet — CHANGES_REQUESTED** [review]
Thanks for the Pull Request!    I've made a couple suggestions for you to investigate.

File: FIRO_TSEnsembles/src/main/resources/database.sql
Comment (danhamill) [reply FIRO_TSEnsembles/src/main/resources/database.sql:1]: Reverted `database.sql` and added `update_20210922_20230718.sql` script to add version to the unique index identifiers.

File: FIRO_TSEnsembles/src/main/java/hec/SqliteDatabase.java (hunk 1 above)
Comment (danhamill) [reply FIRO_TSEnsembles/src/main/java/hec/SqliteDatabase.java]: Abstracted `readEnsembleTimeSeriesFromDb` to work a prepared statement from `RecordIdentifier` or `VersionIdentifier`

File: FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db
Comment (ktarbet) [inline FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db:1]: FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db should not be included in the PR.  That is a temporary test file.

**Review by ktarbet — APPROVED** [review]

File: FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db
Comment (danhamill) [reply FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db:1]: Removed local test file from version tracking

Comment (danhamill) [reply FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db:1]: Added `FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db` to version tracking for build server unit tests.

Comment (ktarbet) [reply FIRO_TSEnsembles/src/test/resources/database/importCsvToDatabase.db:1]: Thanks!   Sorry, I was confused seeing importCsvToDatabase.db in the PR, looks like we need that.
